server/src/main/java/org/opensearch/common/util/ReorganizingLongHash.java

```diff
@@ -118,10 +118,19 @@ public ReorganizingLongHash(final long initialCapacity, final float loadFactor,
         mask = capacity - 1;
         grow = (long) (capacity * loadFactor);
         size = 0;
-
-        table = bigArrays.newLongArray(capacity, false);
-        table.fill(0, capacity, -1);  // -1 represents an empty slot
-        keys = bigArrays.newLongArray(initialCapacity, false);
+        boolean success = false;
+        try {
+            table = bigArrays.newLongArray(capacity, false);
+            table.fill(0, capacity, -1);  // -1 represents an empty slot
+            keys = bigArrays.newLongArray(initialCapacity, false);
+            success = true;
+        } finally {
+            if (!success) {
```

Comment:
Nice finding @neetikasinghal ! I think we could simplify it a bit, when any of the above fail, either `table` or `keys` are `null`, so we don't need the `success` flag here:

```suggestion
            if (table == null || keys == null) {
```